server/src/main/java/org/elasticsearch/index/store/FsDirectoryFactory.java

```diff
@@ -149,7 +149,13 @@ public void close() throws IOException {
             IOUtils.close(super::close, delegate);
         }
 
-        boolean useDelegate(String name) {
+        boolean useDelegate(String name, IOContext ioContext) {
+            if (ioContext == Store.READONCE_CHECKSUM) {
```

Comment:
Nice to see this context useful :)

Reply:
Yep, we had a discussion about #56933 on Slack on Friday and then my brain suddenly realised yesterday that in 7.x we already had all the machinery we need to avoid preloading here.